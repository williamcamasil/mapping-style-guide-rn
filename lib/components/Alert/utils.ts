import { ViewStyle } from 'react-native';

import { Icons } from '../../assets';
import { AppThemeType } from '../../theme/default';
import { ColorPaletesType } from '../../tokens';

export type AlertVariantType =
  'success' |
  'info' |
  'warning' |
  'error';

export const getVariantColorStyle = (variant?: AlertVariantType): ColorPaletesType => {
  switch (variant) {
    case 'warning':
      return 'feedbackAlert500';
    case 'error':
      return 'feedbackError500';
    case 'info':
      return 'neutralGray600';
    case 'success':
    default:
      return 'feedbackSuccess500';
  }
};

export const getDefaultIcon = (variant?: AlertVariantType) => {
  switch (variant) {
    case 'success':
      return Icons.Default.Success;
    case 'info':
      return Icons.Default.Information;
    case 'warning':
    case 'error':
    default:
      return Icons.Default.Warning;
  }
};

export const getAlertStyle = (
  theme: AppThemeType,
  variant?: AlertVariantType,
): ViewStyle => {

  switch (variant) {
    case 'warning':
      return {
        backgroundColor: theme.colors.feedbackAlert100,
      };
    case 'error':
      return {
        backgroundColor: theme.colors.feedbackError100,
      };
    case 'info':
      return {
        backgroundColor: theme.colors.neutralGray100,
      };
    case 'success':
    default:
      return {
        backgroundColor: theme.colors.feedbackSuccess100,
      };
  }
};
