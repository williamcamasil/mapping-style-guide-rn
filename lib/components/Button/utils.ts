import { ViewStyle } from 'react-native';

import { AppThemeType } from '../../theme/default';
import { ColorPaletesType } from '../../tokens';

export type ButtonVariantType =
  'containedPrimary' |
  'containedSecondary' |
  'outlinedPrimary' |
  'outlinedSecondary' |
  'text';

export type ButtonSizeType =
  'large' |
  'medium' |
  'small';

export const getVariantStyle = (
  theme: AppThemeType,
  variant?: ButtonVariantType,
  disabled?: boolean | null,
): ViewStyle => {
  switch (variant) {
    case 'containedSecondary':
      return {
        backgroundColor: disabled ? theme.colors.primary100 : theme.colors.primary200,
        borderWidth: theme.borders.width.none,
      };
    case 'outlinedPrimary':
      return {
        backgroundColor: 'transparent',
        borderWidth: theme.borders.width.thin,
        borderColor: disabled ? theme.colors.primary100 : theme.colors.primary600,
      };
    case 'outlinedSecondary':
      return {
        backgroundColor: 'transparent',
        borderWidth: theme.borders.width.thin,
        borderColor: disabled ? theme.colors.neutralGray100 : theme.colors.neutralGray300,
      };
    case 'text':
      return {
        backgroundColor: disabled ? theme.colors.neutralGray100 : 'transparent',
        borderWidth: theme.borders.width.none,
      };
    case 'containedPrimary':
    default:
      return {
        backgroundColor: disabled ? theme.colors.primary100 : theme.colors.primaryMain,
        borderWidth: theme.borders.width.none,
      };
  }
};

export const getTextColor = (
  variant?: ButtonVariantType,
  disabled?: boolean | null,
): ColorPaletesType => {
  switch (variant) {
    case 'containedSecondary':
      return disabled ? 'primary300' : 'primaryMain';
    case 'outlinedPrimary':
      return disabled ? 'primary300' : 'primaryMain';
    case 'outlinedSecondary':
      return disabled ? 'neutralGray200' : 'neutralGray600';
    case 'text':
      return disabled ? 'neutralGray400' : 'neutralGray500';
    case 'containedPrimary':
    default:
      return disabled ? 'primary300' : 'neutralWhite';
  }
};

export const getSizeStyle = (theme: AppThemeType, size?: ButtonSizeType, hasText?: boolean): ViewStyle => {
  switch (size) {
    case 'large':
      return {
        height: 64,
        minWidth: 64,
        paddingHorizontal: hasText ? theme.spacings.sXL : undefined,
      };
    case 'small':
      return {
        height: 30,
        minWidth: 30,
        paddingHorizontal: hasText ? theme.spacings.sXS : undefined,
      };
    case 'medium':
    default:
      return {
        height: 48,
        minWidth: 48,
        paddingHorizontal: hasText ? theme.spacings.sMedium : undefined,
      };
  }
};

export const getStylesToBlock = (block: boolean | undefined): ViewStyle | null => {
  if (block) return null;

  return {
    flexDirection: 'row',
  };
};
