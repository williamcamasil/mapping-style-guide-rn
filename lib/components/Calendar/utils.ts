import { dateToString } from 'mapping-context-rn';
import type { MarkedDates, Theme } from 'react-native-calendars/src/types';

import { AppThemeType } from '../../theme';

type ExtendedTheme = Theme & {
  'stylesheet.calendar.header': object;
  'stylesheet.day.basic': object;
};

/**
 * Define o tipo de data para o formato yyyy-MM-dd
 */
export type StringDateType = `${number}${number}${number}${number}-${number}${number}-${number}${number}`;

export const DATE_FORMAT = 'yyyy-MM-dd';

export const DAY_SIZE = 42;

export const getDefaultValue = (value?: StringDateType | null) => (
  value ?? dateToString(new Date(), DATE_FORMAT) as StringDateType
);

export const getDisabledDates = (disabledDates?: StringDateType[]) => {
  if (!disabledDates?.length) return {};
  const dates: MarkedDates = {};
  disabledDates.forEach((date: StringDateType) => {
    dates[date] = {
      disabled: true,
      disableTouchEvent: true,
    };
  });
  return dates;
};

export const getCommonTheme = (theme: AppThemeType): ExtendedTheme => ({
  selectedDayBackgroundColor: theme.colors.primaryMain,
  todayTextColor: theme.colors.primaryMain,
  dayTextColor: theme.colors.neutralGray600,
  textDayFontFamily: theme.typography.weights.bold.fontFamily,
  textMonthFontFamily: theme.typography.weights.bold.fontFamily,
  textDayHeaderFontFamily: theme.typography.weights.bold.fontFamily,
  textDayFontWeight: theme.typography.weights.bold.fontWeight,
  textMonthFontWeight: theme.typography.weights.bold.fontWeight,
  textDayHeaderFontWeight: theme.typography.weights.bold.fontWeight,
  textDayFontSize: theme.typography.sizes.small,
  textMonthFontSize: theme.typography.sizes.small,
  textDayHeaderFontSize: theme.typography.sizes.small,
  'stylesheet.calendar.header': {
    dayHeader: {
      marginTop: theme.spacings.sXS,
      marginBottom: theme.spacings.sXS,
      textAlign: 'center',
      fontSize: theme.typography.sizes.small,
      color: theme.colors.primaryMain,
      fontFamily: theme.typography.weights.bold.fontFamily,
      fontWeight: theme.typography.weights.bold.fontWeight,
      flex: 1,
    },
    week: {
      flexDirection: 'row',
      justifyContent: 'space-between',
    },
  },
  'stylesheet.day.basic': {
    base: {
      height: DAY_SIZE,
      width: DAY_SIZE,
      borderBottomLeftRadius: theme.borders.radius.XL,
      borderBottomRightRadius: theme.borders.radius.XL,
      borderTopLeftRadius: theme.borders.radius.XL,
      borderTopRightRadius: theme.borders.radius.XL,
      alignItems: 'center',
      justifyContent: 'center',
    },
  },
});
