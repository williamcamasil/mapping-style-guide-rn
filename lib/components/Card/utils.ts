import { ColorValue, ViewStyle } from 'react-native';

import { AppThemeType } from '../../theme/default';

export type CardSizeType =
  'small' |
  'big';

export type CardVariantType =
  'default' |
  'highlighted50' |
  'highlighted100';

export const getCardSizeStyle = (theme: AppThemeType, size?: CardSizeType): ViewStyle => {
  switch (size) {
    case 'small':
      return {
        width: 135,
        height: 120,
        padding: theme.spacings.sXS,
      };
    case 'big':
    default:
      return {
        minHeight: 128,
        padding: theme.spacings.sSmall,
      };
  }
};

export const getCardShadowStyle = (theme: AppThemeType, showShadow?: boolean): ViewStyle | null => {
  if (showShadow) {
    return theme.shadows.level4;
  }

  return {
    borderWidth: theme.borders.width.thin,
    borderColor: theme.colors.neutralGray200,
  };
};

export const getCardVariantStyle = (theme: AppThemeType, cardType?: CardVariantType): ViewStyle | null => {
  switch (cardType) {
    case 'highlighted50':
      return {
        borderWidth: theme.borders.width.thin,
        borderColor: theme.colors.primary700,
      };
    case 'highlighted100':
      return {
        borderWidth: theme.borders.width.thin,
        borderColor: theme.colors.primary600,
        backgroundColor: theme.colors.primaryMain,
      };
    case 'default':
    default:
      return null;
  }
};

export const getCardCustomBorderStyle = (theme: AppThemeType, borderColor?: ColorValue): ViewStyle | null => {
  if (!borderColor) return null;

  return {
    borderWidth: theme.borders.width.thin,
    borderColor,
  };
};

export const getArrowColor = (theme: AppThemeType, variant?: CardVariantType, arrowColor?: ColorValue): ColorValue => {
  if (arrowColor) return arrowColor;

  if (variant === 'highlighted100') return theme.colors.neutralWhite;

  return theme.colors.neutralGray500;
};
