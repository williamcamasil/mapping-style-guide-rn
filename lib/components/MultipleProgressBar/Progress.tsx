import React, { useMemo } from 'react';
import { StyleProp, ViewStyle } from 'react-native';

import { LinearGradient } from 'expo-linear-gradient';

import { useStyles } from '../../hooks';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';
import { ColorPaletesType } from '../../tokens';

type ProgressPropsType = {
  progress: number;
  theme: AppThemeType;
  startColor: ColorPaletesType;
  endColor: ColorPaletesType;
  style?: StyleProp<ViewStyle>;
};

const startLinearGradient = { x: 0, y: 0 };
const endLinearGradient = { x: 1, y: 0 };

const Progress: React.FC<ProgressPropsType> = ({
  progress, theme, startColor, endColor, style,
}) => {
  const colorGradient = useMemo(
    () => [theme.colors[startColor], theme.colors[endColor]],
    [endColor, startColor, theme],
  );

  const styleBar = useStyles(
    () => [
      {
        flex: progress,
        height: theme.spacings.sNano,
        borderRadius: theme.spacings.sNano / 2,
      },
      style,
    ],
    [progress, style, theme.spacings.sNano],
  );

  return (
    <LinearGradient
      testID="progress-linear-gradient"
      colors={colorGradient}
      start={startLinearGradient}
      end={endLinearGradient}
      style={styleBar}
    />
  );
};

Progress.defaultProps = {
  progress: 0,
};

export default withTheme(Progress);
