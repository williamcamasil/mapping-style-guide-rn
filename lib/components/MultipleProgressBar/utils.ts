
import { ViewStyle } from 'react-native';

import { AppThemeType } from '../../theme';
import { ColorPaletesType } from '../../tokens';

export type MultipleProgressPropsType = {
  colorGradient: [ColorPaletesType, ColorPaletesType];
  value: number;
};

const firstProgressStyles = (theme: AppThemeType): ViewStyle => ({
  borderTopRightRadius: theme.spacings.sZero,
  borderBottomRightRadius: theme.spacings.sZero,
  marginRight: 3,
});

const lastProgressStyles = (theme: AppThemeType): ViewStyle => ({
  borderTopLeftRadius: theme.spacings.sZero,
  borderBottomLeftRadius: theme.spacings.sZero,
});

const middleProgressStyles = (theme: AppThemeType): ViewStyle => ({
  ...firstProgressStyles(theme),
  ...lastProgressStyles(theme),
});

export const getProgressStyles = (theme: AppThemeType, index: number, length: number): ViewStyle | undefined => {
  if (length === 1) return undefined;

  if (index === 0) {
    return firstProgressStyles(theme);
  }

  if (length === (index + 1)) {
    return lastProgressStyles(theme);
  }

  return middleProgressStyles(theme);
};
