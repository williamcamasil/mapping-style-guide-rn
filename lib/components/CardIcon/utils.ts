import { ColorValue } from 'react-native';

import { AppThemeType } from '../../theme/default';
import { CardVariantType } from '../Card/utils';

export const getIconColor = (
  theme: AppThemeType,
  customColor?: ColorValue,
  variant?: CardVariantType,
): ColorValue | string => {
  if (customColor) return customColor;

  if (variant === 'highlighted100') {
    return theme.colors.neutralWhite;
  }

  return theme.colors.primaryMain;
};
