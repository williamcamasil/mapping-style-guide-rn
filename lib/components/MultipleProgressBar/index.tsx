import React from 'react';
import { StyleSheet, View } from 'react-native';

import { useViewStyles } from '../../hooks';
import { AppThemeType, withTheme } from '../../theme';
import { ColorPaletesType } from '../../tokens';
import Progress from './Progress';
import { getProgressStyles, MultipleProgressPropsType } from './utils';

type MultipleProgressBarPropsType = {
  theme: AppThemeType;
  backgroundColor?: ColorPaletesType;
  progress: MultipleProgressPropsType[];
};

const styled = StyleSheet.create({
  container: {
    flexDirection: 'row',
    overflow: 'hidden',
    height: 12,
    padding: 2,
  },
});

const MultipleProgressBar: React.FC<MultipleProgressBarPropsType> = ({
  theme,
  backgroundColor,
  progress,
}) => {
  const styledContainer = useViewStyles(
    () => [
      styled.container,
      {
        borderColor: backgroundColor ?? theme.colors.neutralGray100,
        backgroundColor: backgroundColor ?? theme.colors.neutralGray100,
      },
    ],
    [backgroundColor, theme],
  );

  const renderProgress = () => progress.map((item, index) => {
    const styleToProgress = getProgressStyles(theme, index, progress.length);

    return (
      <Progress
        progress={item.value}
        startColor={item.colorGradient[0]}
        endColor={item.colorGradient[1]}
        style={styleToProgress}
        // eslint-disable-next-line react/no-array-index-key
        key={index}
      />
    );
  });

  return (
    <View testID="container-multiple-progress" style={styledContainer}>
      {renderProgress()}
    </View>
  );
};

export default withTheme(MultipleProgressBar);
