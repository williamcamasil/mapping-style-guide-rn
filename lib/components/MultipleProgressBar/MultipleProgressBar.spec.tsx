import React from 'react';
import { StyleSheet } from 'react-native';

import { render, within } from '@testing-library/react-native';
import { LinearGradient } from 'expo-linear-gradient';

import MultipleProgressBar from '.';
import { DefaultTheme } from '../../theme';
import Progress from './Progress';
import { MultipleProgressPropsType } from './utils';

const multipleProgress: MultipleProgressPropsType[] = [
  {
    value: 1000,
    colorGradient: ['secondary600', 'secondaryMain'],
  },
  {
    value: 500,
    colorGradient: ['primary700', 'primaryMain'],
  },
  {
    value: 500,
    colorGradient: ['feedbackSuccess500', 'feedbackSuccess400'],
  },
  {
    value: 500,
    colorGradient: ['secondary600', 'secondaryMain'],
  },
];

describe('ProgressBar snapshots', () => {
  it('single progress', () => {
    const tree = render(<MultipleProgressBar progress={multipleProgress.slice(0, 1)} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('two progress', () => {
    const tree = render(<MultipleProgressBar progress={multipleProgress.slice(0, 2)} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('multiple progress', () => {
    const tree = render(<MultipleProgressBar progress={multipleProgress} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('ProgressBar tests', () => {
  it('Should render single progress', () => {
    const progressList = [multipleProgress[0]];

    const component = render(<MultipleProgressBar progress={progressList} />);

    const container = within(component.getByTestId('container-multiple-progress'));

    const progressComponent = container.UNSAFE_getAllByType(Progress);

    expect(progressComponent.length).toEqual(1);
  });

  it('Should render single progress and verify colors', () => {
    const progressList = [multipleProgress[0]];

    const component = render(<MultipleProgressBar progress={progressList} />);

    const container = within(component.getByTestId('container-multiple-progress'));

    const progressComponent = container.UNSAFE_getByType(Progress);

    expect(progressComponent.props).toHaveProperty('progress', progressList[0].value);
    expect(progressComponent.props).toHaveProperty('startColor', progressList[0].colorGradient[0]);
    expect(progressComponent.props).toHaveProperty('endColor', progressList[0].colorGradient[1]);
  });

  it('Should render single progress and verify colors default when is undefined in object', () => {
    const progressList = [multipleProgress[multipleProgress.length - 1]];

    const component = render(<MultipleProgressBar progress={progressList} />);

    const container = within(component.getByTestId('container-multiple-progress'));

    const progressComponent = container.UNSAFE_getByType(Progress);
    const linearGradientComponent = container.UNSAFE_getByType(LinearGradient);

    expect(progressComponent.props).toHaveProperty('progress', progressList[0].value);
    expect(progressComponent.props).toHaveProperty('startColor', progressList[0].colorGradient[0]);
    expect(progressComponent.props).toHaveProperty('endColor', progressList[0].colorGradient[1]);

    const colors = [DefaultTheme.colors.secondary600, DefaultTheme.colors.secondaryMain];

    expect(linearGradientComponent.props).toHaveProperty('colors', colors);
  });

  it('Should render two progress with style marginRight separator', () => {
    const progressList = [multipleProgress[0], multipleProgress[1]];

    const component = render(<MultipleProgressBar progress={progressList} />);

    const container = within(component.getByTestId('container-multiple-progress'));

    const progressComponent = container.UNSAFE_getAllByType(Progress);

    const firstProgressStyled = StyleSheet.flatten(progressComponent[0].props.style);
    const lastProgressStyled = StyleSheet.flatten(progressComponent[0].props.style);

    expect(progressComponent.length).toEqual(2);
    expect(firstProgressStyled).toHaveProperty('marginRight', 3);
    expect(lastProgressStyled).toHaveProperty('marginRight', 3);
  });

  it('Should render two progress and verify colors', () => {
    const progressList = [multipleProgress[0], multipleProgress[1]];

    const component = render(<MultipleProgressBar progress={progressList} />);

    const container = within(component.getByTestId('container-multiple-progress'));

    const progressComponent = container.UNSAFE_getAllByType(Progress);

    progressComponent.forEach((progress, index) => {
      expect(progress.props).toHaveProperty('startColor', progressList[index].colorGradient[0]);
      expect(progress.props).toHaveProperty('endColor', progressList[index].colorGradient[1]);
    });
  });
});
