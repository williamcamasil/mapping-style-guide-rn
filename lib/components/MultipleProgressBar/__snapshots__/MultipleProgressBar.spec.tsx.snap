// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar snapshots multiple progress 1`] = `
<View
  style={
    {
      "backgroundColor": "#F7F7F8",
      "borderColor": "#F7F7F8",
      "flexDirection": "row",
      "height": 12,
      "overflow": "hidden",
      "padding": 2,
    }
  }
  testID="container-multiple-progress"
>
  <View
    colors={
      [
        "#F29B72",
        "#E95814",
      ]
    }
    end={
      {
        "x": 1,
        "y": 0,
      }
    }
    start={
      {
        "x": 0,
        "y": 0,
      }
    }
    style={
      {
        "borderBottomRightRadius": 0,
        "borderRadius": 4,
        "borderTopRightRadius": 0,
        "flex": 1000,
        "height": 8,
        "marginRight": 3,
      }
    }
    testID="progress-linear-gradient"
  />
  <View
    colors={
      [
        "#0F4199",
        "#013188",
      ]
    }
    end={
      {
        "x": 1,
        "y": 0,
      }
    }
    start={
      {
        "x": 0,
        "y": 0,
      }
    }
    style={
      {
        "borderBottomLeftRadius": 0,
        "borderBottomRightRadius": 0,
        "borderRadius": 4,
        "borderTopLeftRadius": 0,
        "borderTopRightRadius": 0,
        "flex": 500,
        "height": 8,
        "marginRight": 3,
      }
    }
    testID="progress-linear-gradient"
  />
  <View
    colors={
      [
        "#028E39",
        "#17C55C",
      ]
    }
    end={
      {
        "x": 1,
        "y": 0,
      }
    }
    start={
      {
        "x": 0,
        "y": 0,
      }
    }
    style={
      {
        "borderBottomLeftRadius": 0,
        "borderBottomRightRadius": 0,
        "borderRadius": 4,
        "borderTopLeftRadius": 0,
        "borderTopRightRadius": 0,
        "flex": 500,
        "height": 8,
        "marginRight": 3,
      }
    }
    testID="progress-linear-gradient"
  />
  <View
    colors={
      [
        "#F29B72",
        "#E95814",
      ]
    }
    end={
      {
        "x": 1,
        "y": 0,
      }
    }
    start={
      {
        "x": 0,
        "y": 0,
      }
    }
    style={
      {
        "borderBottomLeftRadius": 0,
        "borderRadius": 4,
        "borderTopLeftRadius": 0,
        "flex": 500,
        "height": 8,
      }
    }
    testID="progress-linear-gradient"
  />
</View>
`;

exports[`ProgressBar snapshots single progress 1`] = `
<View
  style={
    {
      "backgroundColor": "#F7F7F8",
      "borderColor": "#F7F7F8",
      "flexDirection": "row",
      "height": 12,
      "overflow": "hidden",
      "padding": 2,
    }
  }
  testID="container-multiple-progress"
>
  <View
    colors={
      [
        "#F29B72",
        "#E95814",
      ]
    }
    end={
      {
        "x": 1,
        "y": 0,
      }
    }
    start={
      {
        "x": 0,
        "y": 0,
      }
    }
    style={
      {
        "borderRadius": 4,
        "flex": 1000,
        "height": 8,
      }
    }
    testID="progress-linear-gradient"
  />
</View>
`;

exports[`ProgressBar snapshots two progress 1`] = `
<View
  style={
    {
      "backgroundColor": "#F7F7F8",
      "borderColor": "#F7F7F8",
      "flexDirection": "row",
      "height": 12,
      "overflow": "hidden",
      "padding": 2,
    }
  }
  testID="container-multiple-progress"
>
  <View
    colors={
      [
        "#F29B72",
        "#E95814",
      ]
    }
    end={
      {
        "x": 1,
        "y": 0,
      }
    }
    start={
      {
        "x": 0,
        "y": 0,
      }
    }
    style={
      {
        "borderBottomRightRadius": 0,
        "borderRadius": 4,
        "borderTopRightRadius": 0,
        "flex": 1000,
        "height": 8,
        "marginRight": 3,
      }
    }
    testID="progress-linear-gradient"
  />
  <View
    colors={
      [
        "#0F4199",
        "#013188",
      ]
    }
    end={
      {
        "x": 1,
        "y": 0,
      }
    }
    start={
      {
        "x": 0,
        "y": 0,
      }
    }
    style={
      {
        "borderBottomLeftRadius": 0,
        "borderRadius": 4,
        "borderTopLeftRadius": 0,
        "flex": 500,
        "height": 8,
      }
    }
    testID="progress-linear-gradient"
  />
</View>
`;
